### PR TITLE
Release 0.4.2

### PR DESCRIPTION
**Pull request by rib**

### CODE REVIEW

File: android-activity/CHANGELOG.md [hunk 1]
```diff
@@ -3,8 +3,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+
+## [0.4.2] - 2022-02-16
 ### Changed
 - The `Activity.finish()` method is now called when `android_main` returns so the `Activity` will be destroyed ([#67](https://github.com/rust-mobile/android-activity/issues/67))
+- The `native-activity` backend now propagate `NativeWindow` redraw/resize and `ContentRectChanged` callbacks to main loop ([#70](https://github.com/rust-mobile/android-activity/pull/70))
```
Comment (MarijnS95) [inline android-activity/CHANGELOG.md:11]: propagateS?

File: android-activity/CHANGELOG.md [hunk 2]
```diff
@@ -3,8 +3,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+- The `game-activity` implementation of `pointer_index()` was fixed to not always return `0` ([#80](https://github.com/rust-mobile/android-activity/pull/84))
+- Added panic guards, around application's `android_main` and native code that could potentially unwind across a Java FFI boundary ([#68](https://github.com/rust-mobile/android-activity/pull/68))
```
Comment (MarijnS95) [inline android-activity/CHANGELOG.md:13]:
```suggestion
- Added `panic` guards around application's `android_main()` and native code that could potentially unwind across a Java FFI boundary ([#68](https://github.com/rust-mobile/android-activity/pull/68))
```

File: examples/agdk-mainloop/app/build.gradle [hunk 3]
```diff
@@ -32,6 +33,7 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+    namespace 'co.realfit.agdkmainloop'
```
Comment (MarijnS95) [inline examples/agdk-mainloop/app/build.gradle:36]: You wanted to change this, right?

File: examples/agdk-mainloop/src/lib.rs [hunk 4]
```diff
@@ -75,9 +75,32 @@ fn android_main(app: AndroidApp) {
...
+/// Since this is a bare minimum test app we don't depend
+/// on any GPU graphics APIs but we do need to at least
+/// convince Android that we're drawing something and are
+/// responsive, otherwise it will stop delivering input
+/// events to us.
+fn dummy_render(native_window: &ndk::native_window::NativeWindow) {
+    unsafe {
+        let mut buf: ndk_sys::ANativeWindow_Buffer = std::mem::zeroed();
+        let mut rect: ndk_sys::ARect = std::mem::zeroed();
+        ndk_sys::ANativeWindow_lock(
```
Comment (MarijnS95) [inline examples/agdk-mainloop/src/lib.rs:97]: We shouldn't forget to bump these to the new API when `ndk 0.5` drops... I already have a branch that fixes some of the breaking changes and will rebase that to include a copy of https://github.com/rust-mobile/rust-android-examples/pull/7 to prep for it.

File: examples/agdk-mainloop/app/build.gradle (hunk 3 above)
Comment (rib) [reply examples/agdk-mainloop/app/build.gradle:36]: It's fine for now

File: examples/agdk-mainloop/src/lib.rs (hunk 4 above)
Comment (rib) [reply examples/agdk-mainloop/src/lib.rs:97]: Yep, I just sync these examples by copying from rust-android-examples.

**Pull request merged**